src/mcts/search.cc

```diff
@@ -451,35 +456,44 @@ EdgeAndNode Search::GetBestChildWithTemperature(Node* parent,
   }
 
   assert(parent->GetChildrenVisits() > 0);
-  std::vector<float> cumulative_sums;
-  float sum = 0.0;
-  const float n_parent = parent->GetN();
+  std::vector<EdgeAndNode> accepted_edges, filtered_edges;
 
   for (auto edge : parent->Edges()) {
     if (parent == root_node_ && !root_limit.empty() &&
         std::find(root_limit.begin(), root_limit.end(), edge.GetMove()) ==
             root_limit.end()) {
       continue;
     }
+
+    if (edge.GetN() > static_cast<unsigned int>(kTempVisitsFloor)) {
+      accepted_edges.push_back(edge);
+    } else {
+      filtered_edges.push_back(edge);
+    }
+  }
+
+  if (accepted_edges.empty()) {
+    accepted_edges = filtered_edges;
+  }
+
+  assert(accepted_edges.size() != 0);
+
+  const float n_parent = parent->GetN();
+  float sum = 0.0;
+  std::vector<float> cumulative_sums;
+  cumulative_sums.reserve(accepted_edges.size());
+
+  for (const auto& edge : accepted_edges) {
     sum += std::pow(edge.GetN() / n_parent, 1 / temperature);
     cumulative_sums.push_back(sum);
   }
 
-  float toss = Random::Get().GetFloat(cumulative_sums.back());
-  int idx =
+  const float toss = Random::Get().GetFloat(cumulative_sums.back());
+  const int idx =
       std::lower_bound(cumulative_sums.begin(), cumulative_sums.end(), toss) -
       cumulative_sums.begin();
 
-  for (auto edge : parent->Edges()) {
-    if (parent == root_node_ && !root_limit.empty() &&
-        std::find(root_limit.begin(), root_limit.end(), edge.GetMove()) ==
-            root_limit.end()) {
-      continue;
```

Comment:
This root_limit checking code was added in #293 and only touched here.. but @Tilps, wouldn't the visit count for root filtered moves be 0 anyway, so there shouldn't be a need to additionally filter for both with and without temperature best child selection?

Reply:
its true there shouldn't need to be a filter for the with temperature best child selection if searching is done with filtering - but without temperature its required as without temperature is supposed to work with only root node visited - thus filtering is needed to protect against the max policy edge may not be in the filter set.  Seems cleaner to just have the filtering applied consistently even if it does nothing.

Reply:
So should I leave this alone?

Reply:
So should I leave this alone?